ConsistencyManager/ConsistencyManager.swift

```diff
@@ -464,10 +464,27 @@ public class ConsistencyManager {
      I tried doing this with inout instead of making it truly functional, but turns out that inout doesn't work very well.
      Changing to inout helped me about 10%, but after changing to a DictionaryHolder and ArrayHolder, performance was improved ~50x.
      */
-    private func childrenAndListenersForModel(model: ConsistencyManagerModel, modelUpdates: DictionaryHolder<String, ConsistencyManagerModel>, listenersArray: ArrayHolder<ConsistencyManagerListener>) {
+    private func childrenAndListenersForModel(model: ConsistencyManagerModel, modelUpdates: DictionaryHolder<String, [ConsistencyManagerModel]>, listenersArray: ArrayHolder<ConsistencyManagerListener>) {
 
         if let id = model.modelIdentifier {
-            modelUpdates.dictionary[id] = model
+            // Here, we want to store a list of all the projections for a model
+            // Normally, this will just be one element as all models with the same id have the same projection
+            // However, if we have multiple versions of the same model, we want to make sure they are all used to merge a new model
+            let projections: [ConsistencyManagerModel]
```

Comment:
Can you expand on what you mean here? We do have a generic interface - with two new optional methods. I presume you're saying we should have a different API here?